docs/modules/ROOT/pages/reference/extensions/bean-validator.adoc

```diff
@@ -38,3 +38,28 @@ Or add the coordinates to your existing project:
 ----
 
 Check the xref:user-guide/index.adoc[User guide] for more information about writing Camel Quarkus applications.
+
+== Usage
+
+=== Configuring the ValidatorFactory
+
+Implementation of this extension leverages https://quarkus.io/guides/validation[Quarkus extension].
+
+Therefore it is not possible to configure ValidatorFactory by Camel's properties ('constraintValidatorFactory`, `messageInterpolator`, `traversableResolver`, `validationProviderResolver` and `validatorFactory`).
+
+You can configure ValidatorFactory by creation of beans which will be injected into the default validatorFactory (created by Quarkus).
+See https://quarkus.io/guides/validation#hibernate-validator-extension-and-cdi[documentation] for more information.
+
+=== Custom validation groups in native mode
+
+For native mode, it is necessary to register definitions of custom validation groups for reflection.
+To do this, annotate interface definition with `@RegisterForReflection`.
```

Comment:
Is it forcely @RegisterForReflection ? Or could it also work with `quarkus.camel.native.reflection` ?
If all works, maybe the camel quarkus reflection doc can be point like [here](https://camel.apache.org/camel-quarkus/2.7.x/reference/extensions/protobuf.html) and [here](https://github.com/apache/camel-quarkus/blob/main/extensions/protobuf/runtime/src/main/doc/configuration.adoc#serializedeserialize-java-beans-using-json-fields-representation) ?

Reply:
I think any kind of registration for reflection will do, I'll verify it and change doc accordingly.

Reply:
https://quarkus.io/guides/writing-native-applications-tips#register-reflection[documentation] seems to point to the quarkus doc about writing extenion ? not sure it is what camel-quarkus users' need ? maybe `xref:user-guide/native-mode.adoc#reflection` would be better ?